assets/src/edit-story/app/story/useStoryReducer/actions.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Internal dependencies
+ */
+import * as types from './types';
+
+// Exposed actions
+const addPage = ( dispatch ) => ( { page } ) =>
+	dispatch( { type: types.ADD_PAGE, payload: { page, position: null } } );
+
+const addPageAt = ( dispatch ) => ( { page, position } ) =>
+	dispatch( { type: types.ADD_PAGE, payload: { page, position } } );
+
+const deletePage = ( dispatch ) => ( { pageId } ) =>
+	dispatch( { type: types.DELETE_PAGE, payload: { pageId } } );
+
+const deleteCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_PAGE, payload: { pageId: null } } );
+
+const updatePageProperties = ( dispatch ) => ( { pageId, properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageId, properties } } );
+
+const updateCurrentPageProperties = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageId: null, properties } } );
+
+const arrangePage = ( dispatch ) => ( { pageId, position } ) =>
+	dispatch( { type: types.ARRANGE_PAGE, payload: { pageId, position } } );
+
+const setCurrentPage = ( dispatch ) => ( { pageId } ) =>
+	dispatch( { type: types.SET_CURRENT_PAGE, payload: { pageId } } );
+
+const addElements = ( dispatch ) => ( { elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { elements } } );
+
+const addElement = ( dispatch ) => ( { element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { elements: [ element ] } } );
+
+const deleteElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds } } );
+
+const deleteSelectedElements = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: null } } );
+
+const deleteElementById = ( dispatch ) => ( { elementId } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: [ elementId ] } } );
+
+const updateElementsById = ( dispatch ) => ( { elementIds, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds, properties } } );
+
+const updateElementById = ( dispatch ) => ( { elementId, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds: [ elementId ], properties } } );
+
+const updateSelectedElements = ( dispatch ) => ( { properties } ) =>
```

Comment:
One comment here, which is ok to delay until later. But `updateSelectedElements({properties})` only works for side panel. The transforming movables have more complicated rules for updating the properties of elements post transformation, but they are definitely _not_ shared values. For instance:

* Drag: `elements[i].x += deltaX` and `elements[i].y += deltaY`. I.e. the same `deltaX` and `deltaY` offsets for all elements, but different base `x`/`y`.
* Rotate: `elements[i].rotationAngle = angle` plus `elements[i].x += deltaX[i]` and `elements[i].y += deltaY[i]`. I.e. the same angle value, but different offsets `deltaX` and `deltaY`.
* Resize: likely all `width[i]`, `height[i]`, `deltaX[i]` and `deltaY[i]` are different per each element.

It might be just easier to update the selected elements by ID in such cases. Or we could try an updater function that takes existing properties and recalculates the new ones.

Reply:
/cc @miina

Reply:
Maybe `updateElements` would be appropriate for that case. E.g. something like [here](https://github.com/ampproject/amp-wp/pull/3909/files#diff-ee03834cf840c0325129d4839b2e65f0R117-R127).

Reply:
That's a really good point, @dvoytenko. It could definitely make sense to add the option of supplying properties as an object or as a function receiving the old props and returning new ones (that'll be merged in) as a result. Do you want to do this now or can we wait until we have multiple uses for this to see what the best interface is?

Reply:
Definitely ok to wait. It just means there might be more `updateById` vs `updateSelected` used for now.